modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py

```diff
@@ -93,76 +101,50 @@ def apply(self, func, *args, **kwargs):
         -----
         It does not matter if `func` is callable or an ``ray.ObjectRef``. Ray will
         handle it correctly either way. The keyword arguments are sent as a dictionary.
+
+        Note: the function is not applied immediately but is added to the execution tree.
         """
-        log = get_logger()
-        self._is_debug(log) and log.debug(f"ENTER::Partition.apply::{self._identity}")
-        data = self._data
-        call_queue = self.call_queue + [[func, args, kwargs]]
-        if len(call_queue) > 1:
-            self._is_debug(log) and log.debug(
-                f"SUBMIT::_apply_list_of_funcs::{self._identity}"
-            )
-            result, length, width, ip = _apply_list_of_funcs.remote(
-                data, *deconstruct_call_queue(call_queue)
-            )
+        if LazyExec.get():
+            de = DeferredExecution(self._data_ref, func, args, kwargs)
+            return self.__constructor__(de)
+
+        data = self._data_ref
+        if not isinstance(data, DeferredExecution):
+            flat_args = not has_list_or_de(args)
+            flat_kwargs = not has_list_or_de(kwargs.values())
+            if flat_args and flat_kwargs:
+                result = remote_exec_func.remote(func, data, *args, **kwargs)
+                return self.__constructor__(*result)
+            de = DeferredExecution(data, func, args, kwargs, 1, flat_args, flat_kwargs)
```

Comment:
```suggestion
            de = DeferredExecution(
                data,
                func,
                args,
                kwargs,
                num_returns=1,
                flat_args=flat_args,
                flat_kwargs=flat_kwargs,
            )
```

Reply:
```python
                flat_args=flat_args,
                flat_kwargs=flat_kwargs,
```
seems excessively verbose to me.

Reply:
I do not see big excessivness here. We just pass positional args as positional and kwargs as kwargs. It is okay that they have same names. When we pass in a number or a different constant to a method, it is not clear what the number or constant denotes for. The case you have with 1, which denotes num_returns.

Reply:
Changed the arguments order.